src/path_cache.rs

```diff
@@ -938,12 +945,248 @@ impl<N: Neighborhood + Sync> PathCache<N> {
+        // recreate sides in renew
+        for (&cp, sides) in renew.iter() {
+            let mut candidates = PointSet::default();
+            let chunk_index = self.get_chunk_index(cp);
+            let chunk = get_chunk!(self, chunk_index);
+
+            for dir in Dir::all() {
+                if sides[dir.num()] != Renew::No {
+                    chunk.calculate_side_nodes(
+                        dir,
+                        (self.width, self.height),
+                        &get_cost,
+                        self.config,
+                        &mut candidates,
+                    );
+                }
+            }
+
+            // Only include nodes that aren't already part of the map
+            let candidates: PointSet = candidates
+                .iter()
+                .filter(|&&pos| self.nodes.id_at(pos).is_none())
+                .map(|&a| a)
+                .collect();
+
+            if candidates.is_empty() {
+                continue;
+            }
+
+            let all_nodes = &mut self.nodes;
+            let nodes = candidates
+                .into_iter()
+                .map(|p| all_nodes.add_node(p, get_cost(p) as usize))
+                .to_vec();
+
+            if !dirty.contains_key(&cp) {
+                let chunk = get_chunk_mut!(self, chunk_index);
+                chunk.add_nodes(
+                    nodes,
+                    &get_cost,
+                    &self.neighborhood,
+                    &mut self.nodes,
+                    &self.config,
+                );
+            } else {
+                for id in nodes {
+                    let chunk = get_chunk_mut!(self, chunk_index);
+                    chunk.nodes.insert(id);
+                }
+            }
+        }
+
+        println!("time to recreates sides: {:?}", Instant::now() - timer);
+
+        let timer = Instant::now();
+
+        let dirty_indices: hashbrown::HashSet<usize> = dirty
+            .keys()
+            .map(|(x, y)| self.get_chunk_index((*x, *y)))
+            .collect();
+
+        let mid_timer = Instant::now();
+
+        let paths: Vec<HashMap<u32, HashMap<(usize, usize), Path<(usize, usize)>>>> = {
+            use std::sync::Arc;
+            let neighborhood = Arc::new(&self.neighborhood);
+            let all_nodes = Arc::new(&self.nodes);
+
+            self.chunks
+                .par_iter_mut()
+                .enumerate()
+                .filter(|(index, _chunk)| dirty_indices.contains(index))
+                .map(|(_index, chunk)| {
+                    let nodes = chunk.nodes.iter().copied().to_vec();
+                    chunk.nodes.clear();
+
+                    chunk.add_nodes_parallel(nodes, &get_cost, *neighborhood, &all_nodes)
+                })
+                .collect()
+        };
+
+        println!("time to get paths: {:?}", Instant::now() - mid_timer);
+
+        let inner_timer = Instant::now();
+
+        for thing in paths {
+            for (id, thing2) in thing {
```

Comment:
Option 1:
- `paths` -> `all_paths` or leave as `paths`
- `thing` -> `paths` or `paths_in_chunk`
- `id` -> `src_id`
- `thing2` -> `paths_from_src`

Option 2:
Move part of this back into chunk and have `add_nodes_parallel` return a `Vec<(NodeId, NodeId, PathSegment)>` that directly contains the parameters to `self.nodes.add_edge`.
I would prefer Option 2.